### PR TITLE
chore(deps): upgrade operator deps and renovate config to use go 1.21 and ubi 9.4

### PR DESCRIPTION
**Pull request by Fortune-Ndlovu**
## Description
Upgrade the operator to use Go 1.21 and UBI 9.4 to ensure the resulting image is A grade.

## Which issue(s) does this PR fix or relate to
- Fixes #[RHIDP-2646](https://issues.redhat.com//browse/RHIDP-2646) - https://issues.redhat.com/browse/RHIDP-2646 

## PR acceptance criteria
- [x] Tests
- [ ] Documentation
- [ ] If the bundle manifests have been updated, make sure to review the [`rhdh-operator.csv.yaml`](../.rhdh/bundle/manifests/rhdh-operator.csv.yaml) file accordingly

## How to test changes / Special notes to the reviewer
1. Verify the build works by running `make image-build`.
2. Ensure all tests pass by running `make test`.
3. Review the changes to `go.mod`, `Dockerfile`, and renovate config.

### CODE REVIEW

**Review by rm3l — CHANGES_REQUESTED** [review]
Job failed because of two many files changed: https://github.com/janus-idp/operator/actions/runs/9906103267/job/27367027374?pr=398

We are not vendoring at this time (see https://issues.redhat.com/browse/RHIDP-1236?focusedId=24349328&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-24349328), so please remove the  directory.

File: .rhdh/docker/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 #@follow_tag(registry.redhat.io/rhel9/go-toolset:latest)
-FROM registry.access.redhat.com/ubi9/go-toolset:1.20.12-3.1712567214 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21.10-1.1719562237 AS builder
```
Comment (rm3l) [inline .rhdh/docker/Dockerfile:16]:
```suggestion
FROM registry.access.redhat.com/ubi9/go-toolset:1.21.11-2 AS builder
```

There seems to be a new tag available. Can we use the most recent one instead?

**Comment by rm3l** [comment]
/cc @nickboldt 

For additional review.

File: docker/Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 #@follow_tag(registry.redhat.io/rhel9/go-toolset:latest)
-FROM registry.access.redhat.com/ubi9/go-toolset:1.20.12-3.1712567214 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21.10-1.1719562237 AS builder
```
Comment (nickboldt) [inline docker/Dockerfile:16]:
```suggestion
FROM registry.access.redhat.com/ubi9/go-toolset:1.21.11-2 AS builder
```

**Review by nickboldt — APPROVED** [review]

**Pull request merged**